src/ai/backend/client/func/domain.py

```diff
@@ -119,7 +119,10 @@ async def create(
             }
         """
         )
-        query = query.replace("$fields", " ".join(f.field_ref for f in fields))
+        resolved_fields = tuple(
+            f.field_ref if isinstance(f, FieldSpec) else domain_fields[f].field_ref for f in fields
+        )
```

Comment:
Though this is a simple logic, it is repeated many times.
Let's refactor this!

Reply:
refactor this as a new function named `resolve_field()`.
additionally, implement `field_resovler` decorator but it is not used yet.
I will not adapt the decorator if using `resolve_field()` is enough.